src/rest/StreamEndpoints.js

```diff
@@ -60,52 +40,82 @@ export async function createStream(props, apiKey = this.options.apiKey) {
         apiKey,
         {
             method: 'POST',
-            body: JSON.stringify(props),
+            data: props,
         },
     )
     return json ? new Stream(this, json) : undefined
 }
 
 export async function getOrCreateStream(props, apiKey = this.options.apiKey) {
-    let json
-
+    let stream
     // Try looking up the stream by id or name, whichever is defined
     if (props.id) {
-        json = await this.getStream(props.id, apiKey)
+        stream = await this.getStream(props.id, apiKey)
     } else if (props.name) {
-        json = await this.getStreamByName(props.name, apiKey)
+        stream = await this.getStreamByName(props.name, apiKey)
     }
 
     // If not found, try creating the stream
-    if (!json) {
-        json = await this.createStream(props, apiKey)
-        debug('Created stream: %s (%s)', props.name, json.id)
+    if (!stream) {
+        stream = await this.createStream(props, apiKey)
+        debug('Created stream: %s (%s)', props.name, stream && stream.id)
     }
 
-    // If still nothing, throw
-    if (!json) {
-        throw new Error(`Unable to find or create stream: ${props.name}`)
-    } else {
-        return new Stream(this, json)
-    }
+    return stream
+}
+
+export async function updateStream(streamId, props, apiKey = this.options.apiKey) {
+    const json = await authFetch(
+        `${this.options.restUrl}/streams/${streamId}`,
+        apiKey,
+        {
+            method: 'PUT',
+            data: props,
+        },
+    )
+    return json ? new Stream(this, json) : undefined
 }
 
-export function produceToStream(streamObjectOrId, data, apiKey = this.options.apiKey, requestOptions = {}, keepAlive = true) {
+export function deleteStream(streamId, apiKey = this.options.apiKey) {
+    return authFetch(
+        `${this.options.restUrl}/streams/${streamId}`,
+        apiKey,
+        {
+            method: 'DELETE',
+        },
+    )
+}
+
+export function getStreamPermissions(streamId, apiKey = this.options.apiKey) {
+    return authFetch(`${this.options.restUrl}/streams/${streamId}/permissions`, apiKey)
+}
+
+export function detectStreamFields(streamId, apiKey = this.options.apiKey) {
+    return authFetch(`${this.options.restUrl}/streams/${streamId}/detectFields`, apiKey)
+}
+
+export function produceToStream(streamOrId, data, apiKey = this.options.apiKey, requestOptions = {}, keepAlive = true) {
     let streamId
-    if (streamObjectOrId instanceof Stream) {
-        streamId = streamObjectOrId.id
+    if (streamOrId instanceof String || typeof streamOrId === 'string') {
```

Comment:
Should this be `streamOrId instanceof Stream` rather than `String`?

Reply:
Oh I see, yeah IMO I wouldn't even try to handle the `new String()` scenario, those types of values are going to be bad news for almost everything doing any kind of argument checks. `typeof` only IMO.

Reply:
Good to know, I just have a habit of checking a string that way (from some of the earliest "how to check if a value is a string in javascript" google searches)